src/polymer/analyze-properties.ts

```diff
@@ -152,6 +152,13 @@ export function analyzeProperties(
 
     prop.type = esutil.CLOSURE_CONSTRUCTOR_MAP.get(prop.type!) || prop.type;
 
+    if (prop.type && !typeTag) {
```

Comment:
I think lines 153-170 could be factored into a named method just concerned with constructing the right type. The  function name will help document the intention.